include/matrix_assembler.h

```diff
@@ -70,34 +71,46 @@ class MatrixAssembler {
   //! \retval jac_ pointer to Jacobian matrix
   std::shared_ptr<Eigen::SparseMatrix<double>> jac() const { return jac_; }
 
-  //**************************************************************
-  //***** Make all nodal discharge known value, test purpose *****
-  //**************************************************************
-
-  //! Assemble Jacobian matrix
-  void sim_assemble_jacobian();
-
-  //! Apply variables (head and discharge) to nodes and pipes
-  void sim_apply_variables();
-
-  //! Initialize variable vector
-  void sim_assemble_variable_vector();
-
  private:
   //! global nodal id and corresponding nodal pointer
   std::map<Index, std::shared_ptr<pipenetwork::Node>> global_nodes_;
   //! global pipe id and corresponding pipe pointer
   std::map<Index, std::shared_ptr<pipenetwork::Pipe>> global_pipes_;
   //! number of nodes in the network
   unsigned nnode_{0};
+  //! number of known nodes in the network
+  unsigned nnode_known_{0};
```

Comment:
Could we check this isn't > nnode_ and maybe some other value so we can always solve the equation